shell/platform/android/io/flutter/plugin/common/BasicMessageChannel.java

```diff
@@ -145,10 +144,15 @@ public void resizeChannelBuffer(int newSize) {
 
   static void resizeChannelBuffer(
       @NonNull BinaryMessenger messenger, @NonNull String channel, int newSize) {
-    Charset charset = Charset.forName("UTF-8");
-    String messageString = String.format(Locale.US, "resize\r%s\r%d", channel, newSize);
-    ByteBuffer message = ByteBuffer.wrap(messageString.getBytes(charset));
-    messenger.send(CHANNEL_BUFFERS_CHANNEL, message);
+    try {
+      final String content = String.format("resize\r%s\r%d", channel, newSize);
+      final byte[] bytes = content.getBytes("UTF-8");
+      ByteBuffer packet = ByteBuffer.allocateDirect(bytes.length);
+      packet.put(bytes);
+      messenger.send(BasicMessageChannel.CHANNEL_BUFFERS_CHANNEL, packet);
+    } catch (UnsupportedEncodingException e) {
+      Log.e(TAG, "Failed to resize channel buffer named " + channel, e);
+    }
```

Comment:
This seems to be unrelated to the bug filed upstream and intended to support channel names that aren't valid as UTF-8 strings or something. Can it just be removed or perhaps serpated out into its own PR?

Reply:
The `catch` is necessary because `String.getBytes(charset)` is declared as `throws UnsupportedEncodingException`

(In practice the exception will not be thrown because UTF-8 is guaranteed to be supported)

Alternatively, this could continue using `Charset.forName` (which does not throw any checked exceptions)